### PR TITLE
rework some config setting setup

### PR DESCRIPTION
**Pull request by Xkeeper0**
these changes require matching changes in the secrets repository: goonstation/goonstation-secret#171

do not merge this without also merging the other changes. in fact, do not merge this at all without verifying that it works as expected on live servers, god forbid we accidentally deadmin everyone

## About the PR 

this pr rewrites some stuff around how ckey lists are loaded for various things, including mentors, HoS, etc.

it also un-tracks `config/admins.txt` and some other config files, replacing them with "`config/admins.sample.txt`"; the intention is that a server host can make whatever changes they want to the admin/mentor/etc. lists without having to modify tracked files that might cause conflicts in the future

it also moves some configuration files out of `strings/` and into `config/` where they belong. silly config files, you dont belong there

the current goonstation configuraton method seems to be using files in `+secret/` to handle configuration. as such, the new proc for reading these lists will now read both the main `config/` older and the `+secret/config/` files, ensuring that this method still works.

technical list:
- some config files moved from strings/ to config/
- new "xxxx.sample.txt" files in config directory, with some notes explaining what they do
- config.txt, admin.txt etc. now gitignored for conflict-free local editing
- if secrets are enabled, these ckey lists are also read from `+secret/config/`
- removes long-disused rules.html and motd.txt

## Why's this needed? 

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   config/admins.txt
```

### CODE REVIEW

File: code/lists/ckey_configs.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+* This could probably be refactored to config files in general, but
+*/
+/proc/load_config_list(filename, as_ckey = 1, secrets = 0)
+	// "fexists doesn't count symlinks as files"
+	// if (!fexists(filename)))
+	//		throw EXCEPTION("lummox!!!!!!!!!!!!!!")
+
+	var/list/result = list()
+
+	if (!secrets && SECRETS_ENABLED)
```
Comment (ZeWaka) [inline code/lists/ckey_configs.dm:43]: can't use the define like this

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by Wirewraith — APPROVED** [review]
Approval reliant on changes requested by @ZeWaka

File: code/lists/ckey_configs.dm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			if (copytext(line, 1, 2) == "#")
+				// Comments don't count either
+				continue
+
+			result += (as_ckey ? ckey(line) : line)
+
+	catch (var/exception/e)
+		// this might not ever happen depending on what dd_file2list does
+		// in the event of missing files. oh well
+		logDiary("Failed to load config file '[filename]': [e]\n")
```
Comment (pali6) [inline code/lists/ckey_configs.dm:72]:
```suggestion
		logDiary("Failed to load config file '[filename]': [e]\n")
		stack_trace("Failed to load config file '[filename]': [e]")
```
Diary is pretty hidden, this is a fairly important thing that could crash and having it actually be in view-runtimes might help noticing it.

File: config/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Ignore config files, but not the samples
+*.txt
```
Comment (pali6) [inline config/.gitignore:2]: This ignores *all* txt files, right? Wouldn't that affect stuff like `strings/` folder?

Comment (ZeWaka) [reply config/.gitignore:2]: The gitignore is in the folder @pali6

**Pull request merged**